x-pack/plugins/cases/public/components/observables/observable_actions_popover_button.tsx

```diff
@@ -0,0 +1,167 @@
+
+    const panelsToBuild = [
+      {
+        id: 0,
+        title: i18n.OBSERVABLE_ACTIONS,
+        items: mainPanelItems,
+      },
+    ];
+
+    if (permissions.delete) {
+      mainPanelItems.push({
+        name: <EuiTextColor color={'danger'}>{i18n.DELETE_OBSERVABLE}</EuiTextColor>,
+        icon: <EuiIcon type="trash" size="m" color={'danger'} />,
+        onClick: () => {
+          closePopover();
+          onDeletionModalOpen();
+        },
+        disabled: isLoading,
+        'data-test-subj': 'cases-observables-delete-button',
+      });
+    }
+
+    if (permissions.update) {
+      mainPanelItems.push({
+        name: <EuiTextColor>{i18n.EDIT_OBSERVABLE}</EuiTextColor>,
+        icon: <EuiIcon type="pencil" size="m" />,
+        onClick: () => {
+          setShowEditModal(true);
+          closePopover();
+        },
+        disabled: isLoading,
+        'data-test-subj': 'cases-observables-edit-button',
+      });
+    }
+
+    return panelsToBuild;
+  }, [closePopover, isLoading, onDeletionModalOpen, permissions]);
+
+  return (
+    <>
+      <EuiPopover
+        id={`cases-observables-popover-${observable.id}`}
+        key={`cases-observables-popover-${observable.id}`}
+        data-test-subj={`cases-observables-popover-${observable.id}`}
+        button={
+          <EuiButtonIcon
+            onClick={tooglePopover}
+            iconType="boxesHorizontal"
+            aria-label={i18n.OBSERVABLE_ACTIONS}
+            color="text"
+            key={`cases-observables-actions-popover-button-${observable.id}`}
+            data-test-subj={`cases-observables-actions-popover-button-${observable.id}`}
+          />
+        }
+        isOpen={isPopoverOpen}
+        closePopover={closePopover}
+        panelPaddingSize="none"
+        anchorPosition="downLeft"
+      >
+        <EuiContextMenu
+          initialPanelId={0}
+          panels={panels}
+          data-test-subj={'cases-observables-popover-context-menu'}
+        />
+      </EuiPopover>
+      {showDeletionModal && (
+        <DeleteAttachmentConfirmationModal
+          title={i18n.DELETE_OBSERVABLE_CONFIRM}
+          confirmButtonText={i18n.DELETE_OBSERVABLE}
+          onCancel={onCancel}
+          onConfirm={onConfirm}
+        />
+      )}
+      {showEditModal && (
+        <EditObservableModal
+          isLoading={isLoading}
+          handleUpdateObservable={async (updatedObservable) => {
+            const updatedObservableIndex = caseData.observables.findIndex(
+              (_observable) => _observable.id === observable.id
+            );
+            const observables = [...caseData.observables];
+            observables[updatedObservableIndex] = {
+              ...observable,
+              ...updatedObservable,
+            };
+
+            postObservables({
+              observables: observables as ObservablePatchType[],
+              caseId: caseData.id,
+              version: caseData.version,
+            }).then(() => {
```

Comment:
should handle when this call fails probably

Reply:
We should not use this kind of approach to create observables. We should have a dedicated hook using React query like the `useReplaceCustomField`.